flow/project.py

```diff
@@ -3053,6 +3053,7 @@ def _add_parameters(status):
                     # TODO: Fill parameters with empty values (or shared values?)
                     raise ValueError("Cannot show parameters for aggregates.")
                 statepoint = self.open_job(id=aggregate_id).statepoint()
+                document = None
```

Comment:
We should only load the state point if it is needed (and not create a copy of the statepoint via the call operator `job.statepoint()` if only document keys are used).
```suggestion
                job = self.open_job(id=aggregate_id)
                # Cache the job state point and document if used to render status parameters.
                statepoint = None
                document = None
```

Reply:
@b-butler @kidrahahjo I would prefer to avoid putting more complexity to the utility function `dotted_get` and instead just use signac's call operators like `job.statepoint()` and `job.document()` to facilitate caching with this extra check for `None`. This should be performant enough and avoids the need for a separate `cache = {}` object.

Reply:
What we could do to improve `dotted_get` is adopt a pattern more like [this microbenchmark](https://github.com/bdice/python_microbenchmarks/blob/main/microbenchmarks/nested_access.py). I worked on this The best code is the one used in "recurse all" which avoids the re-joining (which probably incurs a memory allocation for the result string) and also avoids the recursion of `dotted_get`. Here is the timing I see for that microbenchmark:
```
      recurse_all: 0.000182753
one_direct_access: 0.000253401
       dotted_get: 0.000997744
```

Reply:
I applied my suggested refactoring in https://github.com/glotzerlab/signac-flow/pull/667/commits/d77269787b4a991a6c1ed18ea8c5244024823cd7. Here's the result for `dotted_get2`, which is the refactored version.

```
      recurse_all: 0.000189232
one_direct_access: 0.00031354
      dotted_get2: 0.000369667
       dotted_get: 0.00102027
```

Reply:
@bdice I wasn't suggesting to change it, just if we were I was providing my preference. I agree with your assessment.